launcher/src/main/java/org/apache/spark/launcher/SparkSubmitOptionParser.java

```diff
@@ -76,6 +76,12 @@
   protected final String PRINCIPAL = "--principal";
   protected final String QUEUE = "--queue";
 
+  // Kubernetes-only options.
+  protected final String KUBERNETES_MASTER = "--kubernetes-master";
```

Comment:
I think I'd prefer a URL like `k8s://host:port/` for referring to the k8s master.  Would be curious to know how other k8s clients refer to the k8s master